app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_counter
```

Comment:
[OPTIONAL]
- You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. It can look something like this 
```
  after_save :update_counter

  private

  def update_counter
    author.increment!(:post_counter)
  end

```

Reply:
good suggestion, I'll apply it in my code